addon/core/rdfa-processor.ts

```diff
@@ -6,41 +7,99 @@ import {
 } from '@lblod/ember-rdfa-editor/utils/_private/dom-helpers';
 import Datastore, {
   EditorStore,
+  SubAndContentPred,
 } from '@lblod/ember-rdfa-editor/utils/_private/datastore/datastore';
 import { Quad } from '@rdfjs/types';
 
-export type ExternalPropertyObject =
-  | {
-      type: 'literal';
-      rdfaId: string;
-    }
-  | {
-      type: 'resource';
-      resource: string;
-    };
-export type ExternalProperty = {
-  type: 'external';
-  predicate: string;
-  object: ExternalPropertyObject;
+export type SayTermType =
+  | 'NamedNode'
+  | 'Literal'
+  | 'LiteralNode'
+  | 'ResourceNode'
+  | 'ContentLiteral'
+  | 'BlankNode';
+
+export type SayRDFLiteral = Pick<
+  RDF.Literal,
+  'value' | 'termType' | 'language'
+> & {
+  datatype: Omit<RDF.NamedNode, 'equals'>;
```

Comment:
What is the main reason for omitting the `equals` method?

Reply:
so we can work with plain POJOs in the forms, having to add the equals method everywhere means we need to use the Dataset constructors from rdfjs everywhere